src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Parser.cs

```diff
@@ -0,0 +1,337 @@
+                            Diag(reportDiagnostic, DiagnosticDescriptors.InvalidLangVersion, methodSyntax.GetLocation());
+                            continue;
+                        }
+
+                        regexMethod.MethodName = regexMethodSymbol.Name;
+                        regexMethod.Modifiers = methodSyntax.Modifiers.ToString();
+                        regexMethod.MatchTimeout ??= Timeout.Infinite;
+                        RegexOptions options = regexMethod.Options.HasValue ? (RegexOptions)regexMethod.Options.Value : RegexOptions.None;
+                        regexMethod.Options = (int)RegexOptions.Compiled | (int)options;
+
+                        // TODO: This is going to include the culture that's current at the time of compilation.
+                        // What should we do about that?  We could say not specifying CultureInvariant is invalid,
+                        // or we could fall back to not generating source if it's not specified, or we could allow
+                        // it and you just get what you get, or we could come up with some culture-aware code gen.
+                        CultureInfo culture = (options & RegexOptions.CultureInvariant) != 0 ? CultureInfo.InvariantCulture : CultureInfo.CurrentCulture;
+
+                        // Validate the options
+                        const RegexOptions SupportedOptions =
+                            RegexOptions.IgnoreCase |
+                            RegexOptions.Multiline |
+                            RegexOptions.ExplicitCapture |
+                            RegexOptions.Compiled |
+                            RegexOptions.Singleline |
+                            RegexOptions.IgnorePatternWhitespace |
+                            RegexOptions.RightToLeft |
+                            RegexOptions.ECMAScript |
+                            RegexOptions.CultureInvariant;
+                        if ((regexMethod.Options.Value & ~(int)SupportedOptions) != 0)
+                        {
+                            Diag(reportDiagnostic, DiagnosticDescriptors.InvalidRegexArguments, methodSyntax.GetLocation(), "options");
+                            continue;
+                        }
+
+                        // Validate the timeout
+                        if (regexMethod.MatchTimeout.Value is 0 or < -1)
+                        {
+                            Diag(reportDiagnostic, DiagnosticDescriptors.InvalidRegexArguments, methodSyntax.GetLocation(), "matchTimeout");
+                            continue;
+                        }
+
+                        // Parse the input pattern
+                        try
+                        {
+                            regexMethod.Tree = RegexParser.Parse(regexMethod.Pattern, (RegexOptions)regexMethod.Options, culture);
+                            regexMethod.Code = RegexWriter.Write(regexMethod.Tree);
+                        }
+                        catch (Exception e)
```

Comment:
any exception? why not just RPE and RMTE? anything else is a bug and it would be nice if it was prominent? (eg don't catch all around the regular regex API)

Reply:
> anything else is a bug and it would be nice if it was prominent? (

It's translating it into a diagnostic.  What would be more prominent?

Reply:
What happens if we just let the exception escape? I guess I mean - is it best practice for all generators to catch all and log a diagnostic? Whatever is the pattern is fine I guess.

Reply:
> What happens if we just let the exception escape?

Escape all the way?  The generator simply stops producing code until you restart VS.